[BUG] `Singular Matrix` error when testing param set with Cross Validation estimators

**Describe the bug**
<!--
A high-level description of what the bug is. Make sure to clearly and concisely describe where the bug was encountered, what happened and what you expected to happen. Provide more details on each of these in the sections below.
-->

It seems that some errors occurs when testing the 3rd paramset with `CoxPH`. Likely due to `CoxPH` not being robust when used on a small dataset. The errors appear as an independent issue when triggering a CI workflow for Python 3.11 with all extras on ubuntu

```
FAILED skpro/tests/test_all_estimators.py::TestAllEstimators::test_fit_does_not_overwrite_hyper_params[RandomizedSearchCV-2-ProbaRegressorSurvival] - numpy.linalg.LinAlgError: Singular matrix
FAILED skpro/tests/test_all_estimators.py::TestAllEstimators::test_fit_updates_state[GridSearchCV-2-ProbaRegressorSurvival] - numpy.linalg.LinAlgError: Singular matrix
FAILED skpro/tests/test_all_estimators.py::TestAllEstimators::test_fit_returns_self[RandomizedSearchCV-2-ProbaRegressorSurvival] - numpy.linalg.LinAlgError: Singular matrix
FAILED skpro/tests/test_all_estimators.py::TestAllEstimators::test_fit_does_not_overwrite_hyper_params[GridSearchCV-2-ProbaRegressorSurvival] - numpy.linalg.LinAlgError: Singular matrix
FAILED skpro/tests/test_all_estimators.py::TestAllEstimators::test_fit_updates_state[RandomizedSearchCV-2-ProbaRegressorSurvival] - numpy.linalg.LinAlgError: Singular matrix
FAILED skpro/tests/test_all_estimators.py::TestAllEstimators::test_fit_returns_self[GridSearchCV-2-ProbaRegressorSurvival] - numpy.linalg.LinAlgError: Singular matrix
```

To reproduce:

Install the skpro package with python3.11 and include `[all_extras]` as well

then call `check_estimator` on both `RandomizedSearchCV` and `GridSearchCV` to generate the error

To fix:

Will generate a bug fix pr to change the estimator from `CoxPH` to a different one and hopefully that resolves the issue

--- a/skpro/model_selection/_tuning.py
+++ b/skpro/model_selection/_tuning.py
@@ -488,8 +488,7 @@ class GridSearchCV(BaseGridSearch):
 
         from skpro.metrics import CRPS, PinballLoss
         from skpro.regression.residual import ResidualDouble
-        from skpro.survival.coxph import CoxPH
-        from skpro.utils.validation._dependencies import _check_estimator_deps
+        from skpro.survival.compose._reduce_cond_unc import ConditionUncensored
 
         linreg1 = LinearRegression()
         linreg2 = LinearRegression(fit_intercept=False)
@@ -510,18 +509,14 @@ class GridSearchCV(BaseGridSearch):
             "error_score": "raise",
         }
 
-        params = [param1, param2]
-
-        # testing with survival predictor
-        if _check_estimator_deps(CoxPH, severity="none"):
-            param3 = {
-                "estimator": CoxPH(alpha=0.05),
-                "cv": KFold(n_splits=4),
-                "param_grid": {"method": ["lpl", "elastic_net"]},
-                "scoring": PinballLoss(),
-                "error_score": "raise",
-            }
-            params.append(param3)
+        params3 = {
+            "estimator": ConditionUncensored(ResidualDouble(LinearRegression())),
+            "cv": KFold(n_splits=4),
+            "param_grid": {"estimator__fit_intercept": [True, False]},
+            "scoring": PinballLoss(),
+            "error_score": "raise",
+        }
+        params = [param1, param2, params3]
 
         return params
 
@@ -747,8 +742,7 @@ class RandomizedSearchCV(BaseGridSearch):
 
         from skpro.metrics import CRPS, PinballLoss
         from skpro.regression.residual import ResidualDouble
-        from skpro.survival.coxph import CoxPH
-        from skpro.utils.validation._dependencies import _check_estimator_deps
+        from skpro.survival.compose._reduce_cond_unc import ConditionUncensored
 
         linreg1 = LinearRegression()
         linreg2 = LinearRegression(fit_intercept=False)
@@ -769,17 +763,13 @@ class RandomizedSearchCV(BaseGridSearch):
             "error_score": "raise",
         }
 
-        params = [param1, param2]
-
-        # testing with survival predictor
-        if _check_estimator_deps(CoxPH, severity="none"):
-            param3 = {
-                "estimator": CoxPH(alpha=0.05),
-                "cv": KFold(n_splits=4),
-                "param_distributions": {"method": ["lpl", "elastic_net"]},
-                "scoring": PinballLoss(),
-                "error_score": "raise",
-            }
-            params += [param3]
+        params3 = {
+            "estimator": ConditionUncensored(ResidualDouble(LinearRegression())),
+            "cv": KFold(n_splits=4),
+            "param_distributions": {"estimator__fit_intercept": [True, False]},
+            "scoring": PinballLoss(),
+            "error_score": "raise",
+        }
+        params = [param1, param2, params3]
 
         return params

